examples/vue3-ssr-vite/src/entry-server.js

```diff
@@ -0,0 +1,67 @@
+import { createApp } from './main';
+import { renderToString } from '@vue/server-renderer';
+
+export async function render(url, manifest) {
+  const { app, router, getResultsState } = createApp({ context: { url } });
+
+  // set the router to the desired URL before rendering
+  router.push(url);
+  await router.isReady();
+
+  // passing SSR context object which will be available via useSSRContext()
+  // @vitejs/plugin-vue injects code into a component's setup() that registers
+  // itself on ctx.modules. After the render, ctx.modules would contain all the
+  // components that have been instantiated during this render call.
+  const ctx = {};
+  const html = await renderToString(app, ctx);
+
+  const algoliaStateScript = `<script>window.__ALGOLIA_STATE__ = ${JSON.stringify(
+    getResultsState()
+  )}</script>`;
```

Comment:
In Vue 2, we did 

```js
context.rendered = () => {
  context.algoliaState = app.instantsearch.getState();
};
```

In Vue 3, we don't have such access like `app.instantsearch`, so `createApp` now returns `getResultsState` function.